onnxruntime/core/providers/cpu/math/inverse.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#include "core/platform/threadpool.h"
+#include "core/util/math_cpuonly.h"
+#include "Eigen/src/Core/Map.h"
+#include "Eigen/LU"
+#include <functional>
+
+namespace onnxruntime {
+class Inverse final : public OpKernel {
+ public:
+  explicit Inverse(const OpKernelInfo& info) : OpKernel(info) {}
+  Status Compute(OpKernelContext* ctx) const override;
+
+ private:
+  template <typename T>
+  struct ComputeImpl;
+};
+
+ONNX_CPU_OPERATOR_KERNEL(
+    Inverse,
+    12,
+    KernelDefBuilder()
+        .TypeConstraint("T", BuildKernelDefConstraints<float, double, MLFloat16>()),
+    Inverse);
+
+template <typename T>
+using MatrixT = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+
```

Comment:
FWIW there's ConstEigenMatrixMapRowMajor and EigenMatrixMapRowMajor in math_cpuonly.h